src/frontend/src/lib/flows/authFlow.svelte.ts

```diff
@@ -107,17 +116,19 @@ export class AuthFlow {
     }
     // Create two try-catch blocks to avoid double-triggering the analytics.
     try {
+      this.authenticating = true;
       const requestConfig = createGoogleRequestConfig(clientId);
       this.systemOverlay = true;
       jwt = await requestJWT(requestConfig, {
         nonce: get(sessionStore).nonce,
         mediation: "required",
       });
     } catch (error) {
-      this.systemOverlay = false;
+      this.authenticating = false;
       this.view = "chooseMethod";
       throw error;
     } finally {
+      this.systemOverlay = false;
```

Comment:
It changes behavior. Now the system overlay is closed after Google popup is closed. However, the UI is still in loading state with the `authenticating` set to true.

This creates a better UX for the user that sees something moving rathern than a static overlay.